src/gui/src/components/explorer-grid/selected-item/context.tsx

```diff
@@ -48,59 +54,103 @@ const getSocketInsights = (
+type SelectedItemStoreState = {
   selectedItem: GridItemData
-  selectedItemDetails: DetailsInfo
   activeTab: Tab
-  setActiveTab: (tab: Tab) => void
-  insights: SocketSecurityDetails[] | undefined
+  selectedItemDetails: DetailsInfo
+  manifest: Manifest | null
   packageScore?: PackageScore
-  manifest?: Manifest | null
+  insights: SocketSecurityDetails[] | undefined
+}
+
+type SelectedItemStoreAction = {
+  setActiveTab: (tab: SelectedItemStoreState['activeTab']) => void
+  setSelectedItemDetails: (
+    selectedItemDetails: SelectedItemStoreState['selectedItemDetails'],
+  ) => void
 }
 
 const SelectedItemContext = createContext<
-  SelectedItemContextValue | undefined
+  | StoreApi<SelectedItemStoreAction & SelectedItemStoreState>
+  | undefined
 >(undefined)
 
+type SelectedItemProviderProps = PropsWithChildren & {
+  selectedItem: GridItemData
+}
+
 export const SelectedItemProvider = ({
   children,
   selectedItem,
-  details,
-  activeTab,
-  setActiveTab,
-}: {
-  children: ReactNode
-  selectedItem: GridItemData
-  details: DetailsInfo
-  activeTab: Tab
-  setActiveTab: (tab: Tab) => void
-}) => {
-  const insights = getSocketInsights(selectedItem.to?.insights)
-  const packageScore = selectedItem.to?.insights.score
-  const manifest = selectedItem.to?.manifest
+}: SelectedItemProviderProps) => {
+  const specOptions = useGraphStore(state => state.specOptions)
 
-  return (
-    <SelectedItemContext.Provider
-      value={{
+  const [selectedItemStore] = useState(() =>
+    createStore<SelectedItemStoreAction & SelectedItemStoreState>(
+      set => ({
         selectedItem,
-        selectedItemDetails: details,
-        activeTab,
-        setActiveTab,
-        insights,
-        packageScore,
-        manifest,
-      }}>
+        manifest: selectedItem.to?.manifest ?? null,
+        packageScore: selectedItem.to?.insights.score,
+        insights: getSocketInsights(selectedItem.to?.insights),
+        selectedItemDetails: {},
+        setSelectedItemDetails: (
+          selectedItemDetails: SelectedItemStoreState['selectedItemDetails'],
+        ) => set(() => ({ selectedItemDetails })),
+        activeTab: 'overview',
+        setActiveTab: (
+          activeTab: SelectedItemStoreState['activeTab'],
+        ) => set(() => ({ activeTab })),
+      }),
+    ),
+  )
+
+  const fetchDetailsAsync = async (
+    store: StoreApi<SelectedItemStoreAction & SelectedItemStoreState>,
+  ) => {
+    const state = store.getState()
+    const item = state.selectedItem
+
+    if (!item.to?.name) return
+
+    const depIdSpec = hydrate(item.to.id, item.to.name, specOptions)
+    const manifest = item.to.manifest ?? {}
+    const abortController = new AbortController()
+
+    for await (const d of fetchDetails(
+      depIdSpec,
+      abortController.signal,
+      manifest,
+    )) {
+      store.setState(state => ({
+        selectedItemDetails: {
+          ...state.selectedItemDetails,
+          ...d,
+        },
+      }))
```

Comment:
In case you agree with the prior suggestion than we will of course need to change it here, along with the `SelectedItemStoreAction` definition and implementation above, just pointing out it here as a reminder.
```suggestion
      store.setState(state => ({
        ...state.selectedItemDetails,
        ...d,
      }))
```